diff --git a/python/docs/ModelMetadataServiceApi.md b/python/docs/ModelMetadataServiceApi.md
index 22abe99..6fe61b7 100644
--- a/python/docs/ModelMetadataServiceApi.md
+++ b/python/docs/ModelMetadataServiceApi.md
@@ -138,13 +138,15 @@ version = 'version_example' # str |  (optional)
 artifact_type = 'UNKNOWN' # str |  (optional) (default to UNKNOWN)
 task_type = 'task_type_example' # str |  (optional)
 model_type = 'model_type_example' # str |  (optional)
+tags = {"tag_key_example": "tag_value_example"} # dict (optional)
+metrics = {"metric_key_example": 123} # dict (optional)
 page_size = 56 # int | Optional. The maximum number of Folders to return in the response. (optional)
 page_token = 'page_token_example' # str | Optional. A pagination token returned from a previous call to `List` that indicates where this listing should continue from. (optional)
 list_mask = 'list_mask_example' # str | Optional. Can be used to specify which fields of Model you wish to return in the response. If left empty all fields will be returned. (optional)
 
 try:
     # List Model Metadata entries.
-    api_response = api_instance.model_metadata_service_list_model_metadata(uri=uri, name=name, version=version, artifact_type=artifact_type, task_type=task_type, model_type=model_type, page_size=page_size, page_token=page_token, list_mask=list_mask)
+    api_response = api_instance.model_metadata_service_list_model_metadata(uri=uri, name=name, version=version, artifact_type=artifact_type, task_type=task_type, model_type=model_type, tags=tags, metrics=metrics, page_size=page_size, page_token=page_token, list_mask=list_mask)
     pprint(api_response)
 except ApiException as e:
     print("Exception when calling ModelMetadataServiceApi->model_metadata_service_list_model_metadata: %s\n" % e)
@@ -160,6 +162,8 @@ Name | Type | Description  | Notes
  **artifact_type** | **str**|  | [optional] [default to UNKNOWN]
  **task_type** | **str**|  | [optional] 
  **model_type** | **str**|  | [optional] 
+ **tags** | **dict**|  | [optional]
+ **metrics** | **dict**|  | [optional]
  **page_size** | **int**| Optional. The maximum number of Folders to return in the response. | [optional] 
  **page_token** | **str**| Optional. A pagination token returned from a previous call to &#x60;List&#x60; that indicates where this listing should continue from. | [optional] 
  **list_mask** | **str**| Optional. Can be used to specify which fields of Model you wish to return in the response. If left empty all fields will be returned. | [optional] 
diff --git a/python/seldon_deploy_sdk/api/model_metadata_service_api.py b/python/seldon_deploy_sdk/api/model_metadata_service_api.py
index d803b4e..05dcf72 100644
--- a/python/seldon_deploy_sdk/api/model_metadata_service_api.py
+++ b/python/seldon_deploy_sdk/api/model_metadata_service_api.py
@@ -243,6 +243,8 @@ class ModelMetadataServiceApi(object):
         :param str artifact_type:
         :param str task_type:
         :param str model_type:
+        :param dict tags:
+        :param dict metrics:
         :param int page_size: Optional. The maximum number of Folders to return in the response.
         :param str page_token: Optional. A pagination token returned from a previous call to `List` that indicates where this listing should continue from.
         :param str list_mask: Optional. Can be used to specify which fields of Model you wish to return in the response. If left empty all fields will be returned.
@@ -273,6 +275,8 @@ class ModelMetadataServiceApi(object):
         :param str artifact_type:
         :param str task_type:
         :param str model_type:
+        :param dict tags:
+        :param dict metrics:
         :param int page_size: Optional. The maximum number of Folders to return in the response.
         :param str page_token: Optional. A pagination token returned from a previous call to `List` that indicates where this listing should continue from.
         :param str list_mask: Optional. Can be used to specify which fields of Model you wish to return in the response. If left empty all fields will be returned.
@@ -281,7 +285,7 @@ class ModelMetadataServiceApi(object):
                  returns the request thread.
         """
 
-        all_params = ['uri', 'name', 'version', 'artifact_type', 'task_type', 'model_type', 'page_size', 'page_token', 'list_mask']  # noqa: E501
+        all_params = ['uri', 'name', 'version', 'artifact_type', 'task_type', 'model_type', 'page_size', 'page_token', 'list_mask', "tags", "metrics"]  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
@@ -320,6 +324,12 @@ class ModelMetadataServiceApi(object):
             query_params.append(('pageToken', params['page_token']))  # noqa: E501
         if 'list_mask' in params:
             query_params.append(('listMask', params['list_mask']))  # noqa: E501
+        if 'tags' in params:
+            for key, val in params["tags"].items():
+                query_params.append((f"tags[{key}]", val))
+        if 'metrics' in params:
+            for key, val in params["metrics"].items():
+                query_params.append((f"metrics[{key}]", val))
 
         header_params = {}

