diff --git a/python/docs/ModelMetadataServiceApi.md b/python/docs/ModelMetadataServiceApi.md
index 150381d..ee5faea 100644
--- a/python/docs/ModelMetadataServiceApi.md
+++ b/python/docs/ModelMetadataServiceApi.md
@@ -138,13 +138,14 @@ version = 'version_example' # str |  (optional)
 implementation = 'implementation_example' # str |  (optional)
 task_type = 'task_type_example' # str |  (optional)
 model_type = 'model_type_example' # str |  (optional)
+tags = {"tag_key_example": "tag_value_example"} # dict (optional)
 query = 'query_example' # str |  (optional)
 page_size = 56 # int | Optional. The maximum number of Folders to return in the response. (optional)
 page_token = 'page_token_example' # str | Optional. A pagination token returned from a previous call to `List` that indicates where this listing should continue from. (optional)
 
 try:
     # List Model Metadata entries.
-    api_response = api_instance.model_metadata_service_list_model_metadata(uri=uri, name=name, version=version, implementation=implementation, task_type=task_type, model_type=model_type, query=query, page_size=page_size, page_token=page_token)
+    api_response = api_instance.model_metadata_service_list_model_metadata(uri=uri, name=name, version=version, implementation=implementation, task_type=task_type, model_type=model_type, tags=tags, query=query, page_size=page_size, page_token=page_token)
     pprint(api_response)
 except ApiException as e:
     print("Exception when calling ModelMetadataServiceApi->model_metadata_service_list_model_metadata: %s\n" % e)
@@ -160,6 +161,7 @@ Name | Type | Description  | Notes
  **implementation** | **str**|  | [optional] 
  **task_type** | **str**|  | [optional] 
  **model_type** | **str**|  | [optional] 
+ **tags** | **dict**|  | [optional]
  **query** | **str**|  | [optional] 
  **page_size** | **int**| Optional. The maximum number of Folders to return in the response. | [optional] 
  **page_token** | **str**| Optional. A pagination token returned from a previous call to &#x60;List&#x60; that indicates where this listing should continue from. | [optional] 
diff --git a/python/seldon_deploy_sdk/api/model_metadata_service_api.py b/python/seldon_deploy_sdk/api/model_metadata_service_api.py
index d6a732c..a3e14b8 100644
--- a/python/seldon_deploy_sdk/api/model_metadata_service_api.py
+++ b/python/seldon_deploy_sdk/api/model_metadata_service_api.py
@@ -243,6 +243,7 @@ class ModelMetadataServiceApi(object):
         :param str implementation:
         :param str task_type:
         :param str model_type:
+        :param dict tags:
         :param str query:
         :param int page_size: Optional. The maximum number of Folders to return in the response.
         :param str page_token: Optional. A pagination token returned from a previous call to `List` that indicates where this listing should continue from.
@@ -273,6 +274,7 @@ class ModelMetadataServiceApi(object):
         :param str implementation:
         :param str task_type:
         :param str model_type:
+        :param dict tags:
         :param str query:
         :param int page_size: Optional. The maximum number of Folders to return in the response.
         :param str page_token: Optional. A pagination token returned from a previous call to `List` that indicates where this listing should continue from.
@@ -281,7 +283,7 @@ class ModelMetadataServiceApi(object):
                  returns the request thread.
         """
 
-        all_params = ['uri', 'name', 'version', 'implementation', 'task_type', 'model_type', 'query', 'page_size', 'page_token']  # noqa: E501
+        all_params = ['uri', 'name', 'version', 'implementation', 'task_type', 'model_type', 'query', 'page_size', 'page_token', "tags"]  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
@@ -320,6 +322,9 @@ class ModelMetadataServiceApi(object):
             query_params.append(('pageSize', params['page_size']))  # noqa: E501
         if 'page_token' in params:
             query_params.append(('pageToken', params['page_token']))  # noqa: E501
+        if 'tags' in params:
+            for key, val in params["tags"].items():
+                query_params.append((f"tags[{key}]", val))
 
         header_params = {}
 
